recipe/patch_yaml/depfinder.yaml

```diff
@@ -0,0 +1,28 @@
+# yaml-language-server: $schema=../patch_yaml_model.json
+# from this bit of code
+# # noarch depfinder packages are broken for python >=3.10
+# if (
+#     record_name == "depfinder"
+#     and record.get("timestamp", 0) < 1659704295850
+#     and subdir == "noarch"
+#     and any(
+#         "<" not in dep
+#         for dep in record.get("depends", [])
+#         if dep.startswith("python ")
+#     )
+# ):
+#     pind = None
+#     for i, dep in enumerate(record.get("depends", [])):
+#         if dep.startswith("python "):
+#             pind = i
+#             break
+
+#     if pind is not None:
+#         record["depends"][pind] = record["depends"][pind] + ",<3.10"
+if:
+  name: depfinder
+  timestamp_lt: 1659704295850
+  subdir: noarch
+  has_depends: "python *[!<]*"
+then:
+  - add_depends: python <3.10  # FIXME? this uses matchspec merge logic
```

Comment:
How do you feel about this type of upper bound capping? It _should_ work just as fine given how `conda` merges `MatchSpec` objects (logical AND if compatible).

Reply:
I think this is fine. My plan had been to make an update to _pin_stricter to handle more cases like this. See this draft PR here: https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/518/files